lib/tasks/batch_push_pom_to_delius.rake

```diff
@@ -0,0 +1,6 @@
+namespace :batch_push_pom_to_delius do
+  desc 'batch push pom to delius job'
+  task populate_delius: :environment do
+    Allocation.where.not(primary_pom_nomis_id: nil) { |a| PushPomToDeliusJob.perform_later(a) }
```

Comment:
In production, this query would return tens of thousands of records.

We should really be using `.find_each` here, to get these records in batches. This will avoid us attempting to load all records at once and exhausting our available memory.

More info about reading records in batches: https://apidock.com/rails/ActiveRecord/Batches/find_each

You can see an example of where batching is performed in another rake task here: https://github.com/ministryofjustice/offender-management-allocation-manager/blob/main/lib/tasks/delius_trigger.rake#L9-L11